sqlglot/generator.py

```diff
@@ -1267,21 +1282,35 @@ def expressions(self, expression, key=None, flat=False, indent=True, sep=", "):
             return ""
 
         if flat:
-            return sep.join(self.sql(e) for e in expressions)
+            return sep.join(self.maybe_comment(self.sql(e), e.comment) for e in expressions)
 
-        sql = (self.sql(e) for e in expressions)
-        # the only time leading_comma changes the output is if pretty print is enabled
+        sql_with_comment = [(self.sql(e), e.comment) for e in expressions]
+        num_sqls = len(sql_with_comment)
```

Comment:
i think this section might be cleaner if we do. right now it seems a bit much

 for i, (sql, comment) in enumerate(sql_with_comment):
    if self.pretty:
         if self._leading_comma:
         else:
    else:
        ....

Reply:
Agreed it seems to much. I'll try that, thanks.